### ISSUE TITLE
[dart2wasm] co19/LanguageFeatures/Extension-methods/accessibility_A02_t02 crash

### ISSUE THREAD

**Issue by sgrekhov** [dart-lang/sdk#56729]
Also `co19/LanguageFeatures/Extension-methods/accessibility_A02_t03`.

```
/===============================================================================================================\
| co19/LanguageFeatures/Extension-methods/accessibility_A02_t02 is new and failed (RuntimeError, expected Pass) |
\===============================================================================================================/

--- Command "dart2js" (took 02.000884s):
DART_CONFIGURATION=ReleaseX64 sdk/bin/dart2js_developer -Dtest_runner.configuration=dart2js-hostasserts-linux-d8 --sound-null-safety --test-mode --packages=/b/s/w/ir/.dart_tool/package_config.json /b/s/w/ir/tests/co19/src/LanguageFeatures/Extension-methods/accessibility_A02_t02.dart --sound-null-safety --out=/b/s/w/ir/out/ReleaseX64/generated_compilations/dart2js-hostasserts-linux-d8/tests_co19_src_LanguageFeatures_Extension-methods_accessibility_A02_t02/accessibility_A02_t02.js

exit code:
0

stdout:
Compiled 10,217,122 input bytes (5,188,003 characters source) to 209,123 characters JavaScript in 2.85 seconds using 265.305 MB of memory

--- Command "d8" (took 104ms):
DART_CONFIGURATION=ReleaseX64 TEST_COMPILATION_DIR=/b/s/w/ir/out/ReleaseX64/generated_compilations/dart2js-hostasserts-linux-d8/tests_co19_src_LanguageFeatures_Extension-methods_accessibility_A02_t02 /b/s/w/ir/third_party/d8/linux/x64/d8 /b/s/w/ir/sdk/lib/_internal/js_runtime/lib/preambles/seal_native_object.js /b/s/w/ir/sdk/lib/_internal/js_runtime/lib/preambles/d8.js /b/s/w/ir/out/ReleaseX64/generated_compilations/dart2js-hostasserts-linux-d8/tests_co19_src_LanguageFeatures_Extension-methods_accessibility_A02_t02/accessibility_A02_t02.js

exit code:
1

stdout:
/b/s/w/ir/sdk/lib/_internal/js_runtime/lib/preambles/d8.js:265: Expect.equals(expected: <foo>, actual: <foo>) fails.
          throw e;
          ^

    at Object.wrapException (/b/s/w/ir/out/ReleaseX64/generated_compilations/dart2js-hostasserts-linux-d8/tests_co19_src_LanguageFeatures_Extension-methods_accessibility_A02_t02/accessibility_A02_t02.js:469:43)
    at Object._fail (/b/s/w/ir/out/ReleaseX64/generated_compilations/dart2js-hostasserts-linux-d8/tests_co19_src_LanguageFeatures_Extension-methods_accessibility_A02_t02/accessibility_A02_t02.js:3668:15)
    at /b/s/w/ir/out/ReleaseX64/generated_compilations/dart2js-hostasserts-linux-d8/tests_co19_src_LanguageFeatures_Extension-methods_accessibility_A02_t02/accessibility_A02_t02.js:3691:17
    at _wrapJsFunctionForAsync_closure.$protected (/b/s/w/ir/out/ReleaseX64/generated_compilations/dart2js-hostasserts-linux-d8/tests_co19_src_LanguageFeatures_Extension-methods_accessibility_A02_t02/accessibility_A02_t02.js:2953:15)
    at _wrapJsFunctionForAsync_closure.call$2 (/b/s/w/ir/out/ReleaseX64/generated_compilations/dart2js-hostasserts-linux-d8/tests_co19_src_LanguageFeatures_Extension-methods_accessibility_A02_t02/accessibility_A02_t02.js:4486:12)
    at _awaitOnObject_closure.call$1 (/b/s/w/ir/out/ReleaseX64/generated_compilations/dart2js-hostasserts-linux-d8/tests_co19_src_LanguageFeatures_Extension-methods_accessibility_A02_t02/accessibility_A02_t02.js:4474:32)
    at _RootZone.runUnary$2$2 (/b/s/w/ir/out/ReleaseX64/generated_compilations/dart2js-hostasserts-linux-d8/tests_co19_src_LanguageFeatures_Extension-methods_accessibility_A02_t02/accessibility_A02_t02.js:4973:18)
    at _Future__propagateToListeners_handleValueCallback.call$0 (/b/s/w/ir/out/ReleaseX64/generated_compilations/dart2js-hostasserts-linux-d8/tests_co19_src_LanguageFeatures_Extension-methods_accessibility_A02_t02/accessibility_A02_t02.js:4888:51)
    at Object._Future__propagateToListeners (/b/s/w/ir/out/ReleaseX64/generated_compilations/dart2js-hostasserts-linux-d8/tests_co19_src_LanguageFeatures_Extension-methods_accessibility_A02_t02/accessibility_A02_t02.js:3139:93)
    at _Future._completeWithValue$1 (/b/s/w/ir/out/ReleaseX64/generated_compilations/dart2js-hostasserts-linux-d8/tests_co19_src_LanguageFeatures_Extension-methods_accessibility_A02_t02/accessibility_A02_t02.js:4736:9)

Deobfuscated error and stack:

    at wrapException                                            org-dartlang-sdk:///lib/_internal/js_runtime/lib/js_helper.dart 1209:19
    at _fail                                                    ../../../../../tests/co19/src/Utils/expect.dart 18:5
    at equals(inlined)                                          ../../../../../tests/co19/src/Utils/expect_common.dart 15:7
    at main                                                     ../../../../../tests/co19/src/LanguageFeatures/Extension-methods/accessibility_A02_t02.dart 21:10
    at _awaitOnObject                                           org-dartlang-sdk:///lib/_internal/js_runtime/lib/async_patch.dart 311:19
    at _wrapJsFunctionForAsync.<anonymous>                      org-dartlang-sdk:///lib/_internal/js_runtime/lib/async_patch.dart 336:23
    at _awaitOnObject.<anonymous>                               org-dartlang-sdk:///lib/_internal/js_runtime/lib/async_patch.dart 287:19
    at _RootZone.runUnary                                       org-dartlang-sdk:///lib/async/zone.dart 1676:46
    at _zone(inlined)                                           org-dartlang-sdk:///lib/async/future_impl.dart 169:29
    at handleValue(inlined)                                     org-dartlang-sdk:///lib/async/future_impl.dart 204:12
    at _Future._propagateToListeners.handleValueCallback        org-dartlang-sdk:///lib/async/future_impl.dart 902:36
    at _Future._propagateToListeners                            org-dartlang-sdk:///lib/async/future_impl.dart 931:13
    at _Future._completeWithValue                               org-dartlang-sdk:///lib/async/future_impl.dart 707:5


--- Re-run this test:
python3 tools/test.py -n dart2js-hostasserts-linux-d8 co19/LanguageFeatures/Extension-methods/accessibility_A02_t02
```
